File: frontend/src/types/animal.ts
export interface Animal {
  id: string;
  numero_arete: string;
  nombre: string | null;
  raza: string | null;
  fecha_nacimiento: string | null;
  fecha_ultimo_parto: string | null;
  activo: boolean;
  baja_fecha: string | null;
  baja_motivo: string | null;
  finca_id: string;
  created_at: string;
  updated_at: string;
}

export interface AnimalCreate {
  numero_arete: string;
  nombre?: string;
  raza?: string;
  fecha_nacimiento?: string;
  fecha_ultimo_parto?: string;
}

export interface AnimalUpdate {
  nombre?: string | null;
  raza?: string | null;
  fecha_nacimiento?: string | null;
  fecha_ultimo_parto?: string | null;
}

File: backend/app/schemas/animal.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class AnimalBase(BaseModel):
    numero_arete: str = Field(..., min_length=1, max_length=50)
    nombre: str | None = Field(None, max_length=100)
    raza: str | None = Field(None, max_length=100)
    fecha_nacimiento: date | None = None
    fecha_ultimo_parto: date | None = None


class AnimalCreate(AnimalBase):
    pass  # finca_id siempre desde el JWT


class AnimalUpdate(BaseModel):
    nombre: str | None = Field(None, max_length=100)
    raza: str | None = Field(None, max_length=100)
    fecha_nacimiento: date | None = None
    fecha_ultimo_parto: date | None = None


class AnimalRead(AnimalBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    finca_id: str
    activo: bool
    baja_fecha: date | None = None
    baja_motivo: str | None = None
    created_at: datetime
    updated_at: datetime
